File: gateway/test/admin-controls-backbone.test.js
const test = require('node:test');
const assert = require('node:assert/strict');

const {
  SERVICE_CONTRACT,
} = require('../src/service-backbones/admin-controls');

test('admin-controls remains a disabled service backbone', () => {
  assert.equal(SERVICE_CONTRACT.lifecycle, 'backbone');
  assert.equal(SERVICE_CONTRACT.enabledByDefault, false);
  assert.equal(SERVICE_CONTRACT.minimumPlan, 'operator');
  assert.ok(SERVICE_CONTRACT.protocolCommands.length > 0);
  assert.ok(SERVICE_CONTRACT.backendMilestones.length > 0);
  assert.ok(SERVICE_CONTRACT.frontendMilestones.length > 0);
  assert.ok(SERVICE_CONTRACT.acceptanceGates.length > 0);
  assert.ok(Object.isFrozen(SERVICE_CONTRACT));
});

File: apps/mobile/test/admin_controls_backbone_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:guardian/services/backbones/admin_controls_backbone.dart';

void main() {
  test('admin-controls stays hidden until its acceptance gates pass', () {
    expect(AdminControlsBackbone.lifecycle, 'backbone');
    expect(AdminControlsBackbone.enabledByDefault, isFalse);
    expect(AdminControlsBackbone.minimumPlan, 'operator');
    expect(AdminControlsBackbone.protocolCommands, isNotEmpty);
    expect(AdminControlsBackbone.frontendMilestones, isNotEmpty);
    expect(AdminControlsBackbone.acceptanceGates, isNotEmpty);
  });
}
